src/main/java/org/myrobotlab/service/abstracts/AbstractMotor.java

```diff
@@ -135,10 +136,29 @@ public AbstractMotor(String n, String id) {
     // input limits
     // "bottom" half of the mapper will be set by the controller
     mapper.map(min, max, -1.0, 1.0);
-    Runtime.getInstance().subscribeToLifeCycleEvents(getName());
-    refreshControllers();
+    registerForInterfaceChange(MotorController.class);
+    // Runtime.getInstance().subscribeToLifeCycleEvents(getName());
+    // refreshControllers();
+  }
+  
+  @Override
+  public void onInterfaceRegistered(String serviceName, String interfaceName) {
+    if (MotorController.class.toString().equals(interfaceName)) {
+      controllers.add(serviceName);

```

Comment:
tracking the list of controllers here probably shouldn't be done, instead some logic like  "if a controller is registered, and it's the one that I am supposed to be attached to,  then the motor should attach to that controller.

Reply:
This is no longer applicable ... I'll update soon

Reply:
removed - now using normalized runtime data